### PR TITLE
Added example for multi device support

### PR DESCRIPTION
**Pull request by pravinraghul**
## Test results
- Tested with two device connected in I2C bus. 
- Device address is 0x50 and 0x52.

> Writing to device 0x50
0x12, 0x34, 0x56, 0x78, 0x9a, 0xbc, 0xde, 0xf0, 
Writing to device 0x52
0x12, 0x34, 0x56, 0x78, 0x9a, 0xbc, 0xde, 0xf0, 
Reading to device 0x50
0x12, 0x34, 0x56, 0x78, 0x9a, 0xbc, 0xde, 0xf0, 
Reading to device 0x52
0x12, 0x34, 0x56, 0x78, 0x9a, 0xbc, 0xde, 0xf0,

### CODE REVIEW

File: examples/MulDevUsage/MulDevUsage.ino [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+void setup()
+{
+
+  uint8_t buffer[8] {};
+
+  Serial.begin(9600);
+
+  eeprom_dev1.begin();
+  eeprom_dev2.begin();
```
Comment (WifWaf) [inline examples/MulDevUsage/MulDevUsage.ino:49]:
```suggestion
  eeprom.begin();
```

**Review by WifWaf — CHANGES_REQUESTED** [review]
Thanks for doing this.

I think I confused things with regard to how the library changes affect usage.

We can use a single instance and switch between addresses with the setDeviceAddress() function.

For multiple instances, it's usually best to add an argument to the constructor of the library, and then add the addresses when instancing (e.g. eeprom_dev1(0x52). This is common in Arduino, so I'm happy to add this to the library as an alternative to setDeviceAddress().

I've added some recommended changes to the library to explain, however, I can't test the library, so if you can test it, that would be great.

File: examples/MulDevUsage/MulDevUsage.ino [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  write(eeprom_dev1); // 0x50
+
+  Serial.println("Writing to device 0x52");
+  write(eeprom_dev2); // 0x52
+
+  Serial.println("Reading to device 0x50");
+  read(eeprom_dev1);  // 0x50
+
+  Serial.println("Reading to device 0x52");
+  read(eeprom_dev2);  // 0x52
```
Comment (WifWaf) [inline examples/MulDevUsage/MulDevUsage.ino:64]:
```suggestion
  Serial.println("Writing to device 0x50");
  write(eeprom); // 0x50 (library defaults to 0x50)

  eeprom.setDeviceAddress(ATDEVADR1);
  Serial.println("Writing to device 0x52");
  write(eeprom); // 0x52
  
  eeprom.setDeviceAddress(ATDEVADR0);
  Serial.println("Reading to device 0x50");
  read(eeprom);  // 0x50
  
  eeprom.setDeviceAddress(ATDEVADR1);
  Serial.println("Reading to device 0x52");
  read(eeprom);  // 0x52
```

Comment (WifWaf) [reply examples/MulDevUsage/MulDevUsage.ino:64]: Just using the single library and switch the addresses here.

**Review by WifWaf — CHANGES_REQUESTED** [review]

File: examples/MulDevUsage/MulDevUsage.ino [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    Serial.printf("0x%x, ", buff[i]);
+  }
+  Serial.println();
+}
+
+
+void setup()
+{
+
+  uint8_t buffer[8] {};
```
[the remark this answers is not in the document]
Comment (pravinraghul) [reply examples/MulDevUsage/MulDevUsage.ino:44]: A buffer array to get the data from the 'readByteArray()'

>  uint8_t buff[8] {};
  eeprom.readByteArray(0, buff, sizeof(buff));

**Comment by pravinraghul** [comment]
Thanks, Using single instance will be simple and good.

I've added the changed to the example.

File: examples/MulDevUsage/MulDevUsage.ino (hunk 3 above)
[the remark this answers is not in the document]
Comment (WifWaf) [reply examples/MulDevUsage/MulDevUsage.ino:44]: See line 44 

 "uint8_t buffer[8] {};"
 
 I think perhaps it was left over from the previous example. :)

**Review by WifWaf — CHANGES_REQUESTED** [review]

File: examples/MulDevUsage/MulDevUsage.ino (hunk 3 above)
[the remark this answers is not in the document]
Comment (pravinraghul) [reply examples/MulDevUsage/MulDevUsage.ino:44]: > A buffer array to get the data from the 'readByteArray()'
> 
> >  uint8_t buff[8] {};
>   eeprom.readByteArray(0, buff, sizeof(buff));
> 

I'm not sure, how I suppose to use this readByteArray() method without a char array pointer.

Can you be more specific?

[the remark this answers is not in the document]
Comment (WifWaf) [reply examples/MulDevUsage/MulDevUsage.ino:44]: You have a buffer declared at line 30 which is passed to the readByteArray() function, while the buffer at line 44 is not used.

[the remark this answers is not in the document]
Comment (pravinraghul) [reply examples/MulDevUsage/MulDevUsage.ino]: Thanks @WifWaf, removed the unused variable

**Pull request merged**